### ISSUE TITLE
[Feature]: utter-project/mHuBERT-147

### ISSUE THREAD

**Issue by Mixomo** [IAHispano/Applio#770]
### Description

mHuBERT-147 are compact and competitive multilingual HuBERT models trained on 90K hours of open-license data in 147 languages. Different from traditional HuBERTs, mHuBERT-147 models are trained using faiss IVF discrete speech units. Training employs a two-level language, data source up-sampling during training.

### Problem

Maybe this is useful to improve RVC multilingual pronunciation?

### Proposed Solution

https://huggingface.co/utter-project/mHuBERT-147

### Alternatives Considered

-

**Comment by AznamirWoW** [comment]
You are free to download this embedder and try it as a custom one to train and then infer an audio.

**Comment by AznamirWoW** [comment]
did 50 epochs with a test set

![image](https://github.com/user-attachments/assets/de56be95-9b21-4134-8076-676674d4340a)

**Comment by Mixomo** [comment]
> did 50 epochs with a test set. Orange is mHuBERT, red is ContentVec
> 
> ![image](https://private-user-images.githubusercontent.com/101997116/372658404-de56be95-9b21-4134-8076-676674d4340a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc4MTg5MjIsIm5iZiI6MTcyNzgxODYyMiwicGF0aCI6Ii8xMDE5OTcxMTYvMzcyNjU4NDA0LWRlNTZiZTk1LTliMjEtNDEzNC04MDc2LTY3NjY3NGQ0MzQwYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAwMVQyMTM3MDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zOWU4N2Q1YTg0MzFjOTRmMjdmZGIzYmExZTRhYWUwYzY5YzY5YTI2MmRkMmIxMDZiZWRhOGNkOTIzYTY2NzQwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.0TRj-9DqznNZFmiZ0e08ZthbdWVJ3BGg3OuRfBv5_jQ)

Great! I'll try it as soon as I can. 
I'm very interested in the Spanish language, what do you think about this one?  
https://huggingface.co/espnet/mediaspeech-spanish-hubert/tree/main

I remember some time ago I wanted to add it as custom, but it gave me a dict error. 

On the other hand, did you have to do something specific to make the Hubert I mentioned above work? Or just selecting it as custom in RVC made it work? Did you use the checkpoint_best.pth or the .bin?

**Comment by AznamirWoW** [comment]
> > did 50 epochs with a test set. Orange is mHuBERT, red is ContentVec
> > ![image](https://private-user-images.githubusercontent.com/101997116/372658404-de56be95-9b21-4134-8076-676674d4340a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc4MTg5MjIsIm5iZiI6MTcyNzgxODYyMiwicGF0aCI6Ii8xMDE5OTcxMTYvMzcyNjU4NDA0LWRlNTZiZTk1LTliMjEtNDEzNC04MDc2LTY3NjY3NGQ0MzQwYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAwMVQyMTM3MDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zOWU4N2Q1YTg0MzFjOTRmMjdmZGIzYmExZTRhYWUwYzY5YzY5YTI2MmRkMmIxMDZiZWRhOGNkOTIzYTY2NzQwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.0TRj-9DqznNZFmiZ0e08ZthbdWVJ3BGg3OuRfBv5_jQ)
> 
> Great! I'll try it as soon as I can. I'm very interested in the Spanish language, what do you think about this one? https://huggingface.co/espnet/mediaspeech-spanish-hubert/tree/main
> 
> I remember some time ago I wanted to add it as custom, but it gave me a dict error.
> 
> On the other hand, did you have to do something specific to make the Hubert I mentioned above work? Or just selecting it as custom in RVC made it work? Did you use the checkpoint_best.pth or the .bin?

That spanish model has no config and no bin.

I simply downloaded the other one (pytorch_model.bin and config.json) and provided them to the custom embedder prompt.

**Comment by Mixomo** [comment]
Ah, you are right, the config is missing, that's a shame... 
Thank you for your answer.

**Comment by Mixomo** [comment]
Starting embedding extraction with 20 cores on cuda:0...

Custom embedder not found: ['rvc\\models\\embedders\\embedders_custom\\custom_hubert_test'], using contentvec

![image](https://github.com/user-attachments/assets/7f3c3dd6-8003-4669-bd63-5fe8b2061fe2)

**Comment by Mixomo** [comment]
> Starting embedding extraction with 20 cores on cuda:0...
> 
> Custom embedder not found: ['rvc\models\embedders\embedders_custom\custom_hubert_test'], using contentvec
> 
> ![image](https://private-user-images.githubusercontent.com/108157310/373094989-ee737d80-4688-469b-8ad3-bce5c81a34a9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc5MjM0MjgsIm5iZiI6MTcyNzkyMzEyOCwicGF0aCI6Ii8xMDgxNTczMTAvMzczMDk0OTg5LWVlNzM3ZDgwLTQ2ODgtNDY5Yi04YWQzLWJjZTVjODFhMzRhOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAwM1QwMjM4NDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lODZjNzlkOGUyM2ZiMzEwZjkyNWMwYWMyOWQxZmI3YmY4OTJlMDg3ZTBkNTcwYThkMTYwYzdiOWRhODhmMTYyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.eGO_cTVnpJLmKfrPJtfwmuS5pPIAoj4Uz__dvYNHOhE)

Fixed, I had to use the full path.

Instead of using `rvc\models\embedders\embedders_custom\custom_hubert_test`

I had to use `J:\Applio V3\ApplioV3.2.6\rvc\models\embedders\embedders_custom\custom_hubert_test`
